src/ext/java/org/opentripplanner/ext/bikerentalservicedirectory/BikeRentalServiceDirectoryFetcher.java

```diff
@@ -0,0 +1,75 @@
+package org.opentripplanner.ext.bikerentalservicedirectory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.opentripplanner.updater.GraphUpdater;
+import org.opentripplanner.updater.bike_rental.BikeRentalUpdater;
+import org.opentripplanner.util.HttpUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Fetches GBFS endpoints from the Bikeservice component located at
+ * https://github.com/entur/bikeservice, which is an API for aggregating GBFS endpoints.
+ */
+public class BikeRentalServiceDirectoryFetcher {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BikeRentalServiceDirectoryFetcher.class);
+
+  private static final int DEFAULT_FREQUENCY_SEC = 15;
+
+  private static final String GBFS_JSON_FILENAME = "gbfs.json";
+
+  public static List<GraphUpdater> createUpdatersFromEndpoint(URI url) {
+
+    LOG.info("Fetching list of updaters from {}", url);
+
+    List<GraphUpdater> updaters = new ArrayList<>();
+
+    try {
+      InputStream is = HttpUtils.getData(url.toString());
```

Comment:
The getData method should be refactored to take in a type-safe argument when we have it. I will fix this.